ui/api/consumerGroups/schema.ts

```diff
@@ -65,3 +108,13 @@ export const ConsumerGroupResponseSchema = z.object({
 export type ConsumerGroupResponse = z.infer<
   typeof ConsumerGroupsResponseSchema
 >;
+
+export type UpdateConsumerGroupErrorSchema = z.infer<
+  typeof UpdateConsumerGroupErrorSchema
+>;
+
+export const ConsumerGroupDryrunResponseSchema = z.object({
```

Comment:
This should be the same schema as `ConsumerGroupResponse`. It looks like all the optional fields are already correctly set in `ConsumerGroupResponse`. Can we use a single schema definition instead?

Reply:
I checked the `ConsumerGroupSchema` before implementing it and noticed it contains several fields that are not required for the dry run. For example:

typescript
Copy code
offsets: z.array(OffsetAndMetadataSchema).optional()
```
const OffsetAndMetadataSchema = z.object({
    topicId: z.string(),
    topicName: z.string(),
    partition: z.number(),
    offset: z.number(),
    logEndOffset: z.number().optional(),
    lag: z.number().optional(),
    metadata: z.string(),
    leaderEpoch: z.number().optional()
});
```
Fields like offset, logEndOffset, and lag are not needed for the dry run, so I created a new response schema for dryrun

Reply:
Right, the dry run will return a subset of the data that you would get from the describe/fetch operator. The back-end API uses the same data model for both. Just mentioning it because it might help reduce maintenance in the future and make it clear that they are the same response schema from the API.